### PR TITLE
[OMNIML-2244] enable fp8 and int8 ONNX export

### PR DESCRIPTION
**Pull request by ajrasane**
## What does this PR do?

**Type of change:** 
Example update

**Overview:**
- Support ONNX export for fp8 and int8 precisions
- Added utility functions to check for fp8 and int8 quantization (will be used in ONNXExporter)
- Fixed a bug in evaluation API for high batch sizes
- Added function to replace zeros from scales to smallest positive value in fp16

## Usage

```python
python torch_quant_to_onnx.py \
    --quantize_mode fp8/int8 \
    --onnx_save_path <onnx_path>  
```

## Testing
Validated the accuracy and latency of int8 and fp8 models:
| Metric | INT8 | FP8 | FP16
|--------|------|-----|------|
| Top1 Accuracy | 84.584% | 85.062% | 85.102%
| Top5 Accuracy | 97.3% | 97.534% | 97.526%

## Before your PR is "*Ready for review*"

- **Make sure you read and follow [Contributor guidelines](https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/main/CONTRIBUTING.md)** and your commits are signed.
- **Is this change backward compatible?**: Yes
- **Did you write any new necessary tests?**: No
- **Did you add or update any necessary documentation?**: Yes
- **Did you update [Changelog](https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/main/CHANGELOG.rst)?**: No

### CODE REVIEW

File: modelopt/torch/_deploy/utils/torch_onnx.py [hunk 1]
```diff
@@ -336,6 +337,32 @@ def is_mxfp8_quantized(model: nn.Module) -> bool:
     return False
 
 
+def is_int8_quantized(model: nn.Module) -> bool:
```
Comment (cjluo-nv) [inline modelopt/torch/_deploy/utils/torch_onnx.py:340]: do we still consider mixed precision here?

Comment (ajrasane) [reply modelopt/torch/_deploy/utils/torch_onnx.py:340]: This MR targets models with only FP8 and INT8 quantization. For mixed precision, I will update the following [function](https://github.com/NVIDIA/TensorRT-Model-Optimizer/pull/575/files#diff-72f915aa60371da0faed1db2ae22db62375e21064973400882ea328eb34e7bbcR340) with this utility function.

File: modelopt/onnx/quantization/qdq_utils.py [hunk 2]
```diff
@@ -1037,6 +1037,21 @@ def remove_graph_input_q(onnx_model: onnx.ModelProto) -> onnx.ModelProto:
     return onnx_model
 
 
+def replace_zero_scale_with_smallest_nonzero(onnx_model: onnx.ModelProto) -> onnx.ModelProto:
+    """Replace zero scale values with smallest nonzero fp16 value in the ONNX model."""
```
Comment (cjluo-nv) [inline modelopt/onnx/quantization/qdq_utils.py:1041]: could you document in what condition do we need to call this method here?

Comment (ajrasane) [reply modelopt/onnx/quantization/qdq_utils.py:1041]: Sorry, I had already set the MR to auto merge. Will add this in a follow up MR.
